module-4/backend/week_four.md

```diff
@@ -4,17 +4,65 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+    icebox
+    current iteration
+    up next
+    done
+* What are the four types of cards available in Pivotal Tracker? 
+    started 
+    finished 
+    delivered 
+    accepted
 * What's your favorite project management tool?
-* Why do we create staging environments?
+    Waffle :3
+* Why do we create staging environments? 
+    so that we can work with the production environment (different api keys + variable names + databases) in a way that does      
+    not effect production in anyway. 
 * What are the characteristics of a good README (in your opinion)?
-* What's one main improvement you're going to make to your code regarding accessibility issues?
-* What are some basic security concerns to be aware of when building applications?
-* Why is continuous integration helpful/important?
+    - A good explaination of how you use the tool 
+    - Listing any endpoints and how you interact with them 
+    - Making it begginner friendly 
+    - Explains how the application was build, what tools were used
+* What's one main improvement you're going to make to your code regarding accessibility issues? 
+    - Make sure you can read through the whole of the available pages and the text to voice program pick up 
+      all the intended text 
+    - Making the webpages calm and minimal as possible
+* What are some basic security concerns to be aware of when building applications? 
+    - How you're setting up the login/logout process -> 
+            
+            - don't store passwords in plain text
+            - at registration that users make passwords strong
+            - using strong params when using incoming data
+            - sanitizing the incoming data 
+            - keeping certain information secure -> whether some pages do or do not exist, if the login was incorrect 
+              revealing which ones were wrong
+          
+* Why is continuous integration helpful/important? 
+    contineous integration is helpful so that you can test the application in a different environment
 * What are some continuous integration tools?
+    Travis CI
 
 #### Review  
 
 * What are some characteristics of "good" git workflow?
+    small commit messages
+    commiting often
+    not working on master
+    having a clear flow of branches merging back into master
 * What are the four fundamental concepts of object oriented programming?
+    Encapsulation - puting things into small functional parts
+    Abstraction - keeping the current file/method as small as possible and taking out anything that doesn't do what the name 
+          says it does 
+    Inheritance - files can inherit from other files. things don't need to be in the same place
+    Writing it once, using it over and over
 * What's a module in Ruby and what's the difference between a class and a module?
+    classes inherit from modules. Modules are the blueprint of how something might look and act but class inherit it and use 
```

Comment:
I would do some more research on this. Classes tend to `extend` or `include` modules. Modules are stateless, and typically provide shared functions that many classes can use.